packages/sdk-cli/src/api/baseAPI.ts

```diff
@@ -1,7 +1,35 @@
+import * as t from 'io-ts';
+
 import * as auth from '../auth';
 import environment from '../auth/environment';
 import fetch, { Headers } from '../fetch';
-import { decodeJSON } from '../util/fetchUtil';
+import { assertJSON } from '../util/fetchUtil';
+
+// tslint:disable-next-line:variable-name
+const ErrorResponse = t.type(
+  {
+    errors: t.array(t.type({
+      message: t.string,
+    })),
+  },
+  'Hosts403Response',
+);
+type ErrorResponse = t.TypeOf<typeof ErrorResponse>;
+
+export const okOrElse = <T>(failure: (response: Response) => Promise<T>) => (response: Response) =>
+  response.ok ? Promise.resolve(response) : failure(response);
+export const assertOK = okOrElse((response) => {
+  return assertJSON()(response)
+    .catch(() => Promise.reject(
+      // tslint:disable-next-line:max-line-length
+      new Error(`Fetch of ${response.url} returned status ${response.status} ${response.statusText}`),
+    ))
+    .then(responseObject => ErrorResponse.decode(responseObject).getOrElseL(() => {
```

Comment:
This is a perfect place to use `fold`.
```suggestion
    .then(responseObject => Promise.reject(new Error(ErrorResponse.decode(responseObject).fold(
      () => `fetch of ${response.url} returned status ${response.status} ${response.statusText} with body: ${JSON.stringify(responseObject, undefined, 2)}`,
      errorObj => errorObj.errors.map(err => err.message).join('\n'),
    )));
```

Reply:
Was trying to find a function like this. Wish they had better documented this stuff.